main/main.c

```diff
@@ -1080,114 +1092,16 @@ static void php_error_cb(int type, const char *error_filename, const uint error_
-					if (type == E_ERROR || type == E_PARSE) {
-						zend_string *buf = php_escape_html_entities((unsigned char*)buffer, buffer_len, 0, ENT_COMPAT, NULL);
-						php_printf("%s<br />\n<b>%s</b>:  %s in <b>%s</b> on line <b>%d</b><br />\n%s", STR_PRINT(prepend_string), error_type_str, buf->val, error_filename, error_lineno, STR_PRINT(append_string));
-						zend_string_free(buf);
-					} else {
-						php_printf("%s<br />\n<b>%s</b>:  %s in <b>%s</b> on line <b>%d</b><br />\n%s", STR_PRINT(prepend_string), error_type_str, buffer, error_filename, error_lineno, STR_PRINT(append_string));
-					}
-				} else {
-					/* Write CLI/CGI errors to stderr if display_errors = "stderr" */
-					if ((!strcmp(sapi_module.name, "cli") || !strcmp(sapi_module.name, "cgi")) &&
-						PG(display_errors) == PHP_DISPLAY_ERRORS_STDERR
-					) {
-#ifdef PHP_WIN32
-						fprintf(stderr, "%s: %s in %s on line %u\n", error_type_str, buffer, error_filename, error_lineno);
-						fflush(stderr);
-#else
-						fprintf(stderr, "%s: %s in %s on line %u\n", error_type_str, buffer, error_filename, error_lineno);
-#endif
-					} else {
-						php_printf("%s\n%s: %s in %s on line %d\n%s", STR_PRINT(prepend_string), error_type_str, buffer, error_filename, error_lineno, STR_PRINT(append_string));
-					}
-				}
-			}
-		}
-#if ZEND_DEBUG
-		if (PG(report_zend_debug)) {
-			zend_bool trigger_break;
-
-			switch (type) {
-				case E_ERROR:
-				case E_CORE_ERROR:
-				case E_COMPILE_ERROR:
-				case E_USER_ERROR:
-					trigger_break=1;
-					break;
-				default:
-					trigger_break=0;
-					break;
-			}
-			zend_output_debug_string(trigger_break, "%s(%d) : %s - %s", error_filename, error_lineno, error_type_str, buffer);
-		}
-#endif
+		CALL_ERROR_HOOKS(E_HOOK_LOG);
+		CALL_ERROR_HOOKS(E_HOOK_DISPLAY);
+		CALL_ERROR_HOOKS(E_HOOK_PROCESS);
 	}
 
-	/* Bail out if we can't recover */
-	switch (type) {
-		case E_CORE_ERROR:
-			if(!module_initialized) {
-				/* bad error in module startup - no way we can live with this */
-				exit(-2);
-			}
-		/* no break - intentionally */
-		case E_ERROR:
-		case E_RECOVERABLE_ERROR:
-		case E_PARSE:
-		case E_COMPILE_ERROR:
-		case E_USER_ERROR:
-			EG(exit_status) = 255;
-			if (module_initialized) {
-				if (!PG(display_errors) &&
-				    !SG(headers_sent) &&
-					SG(sapi_headers).http_response_code == 200
-				) {
-					sapi_header_line ctr = {0};
-
-					ctr.line = "HTTP/1.0 500 Internal Server Error";
-					ctr.line_len = sizeof("HTTP/1.0 500 Internal Server Error") - 1;
-					sapi_header_op(SAPI_HEADER_REPLACE, &ctr);
-				}
-				/* the parser would return 1 (failure), we can bail out nicely */
-				if (type == E_PARSE) {
-					CG(parse_error) = 0;
-				} else {
-					/* restore memory limit */
-					zend_set_memory_limit(PG(memory_limit));
-					efree(buffer);
-					zend_objects_store_mark_destructed(&EG(objects_store));
-					zend_bailout();
-					return;
-				}
-			}
-			break;
-	}
+	CALL_ERROR_HOOKS(E_HOOK_BAILOUT);
```

Comment:
This passes error_type_str which is initialized only if display is true and other conditions above are fulfilled, so in case they are not, you get uninitialized pointer there.